src/mapper.rs

```diff
@@ -293,6 +294,12 @@ impl<'s> ProguardMapper<'s> {
         self.classes.get(class).map(|class| class.original)
     }
 
+    /// returns a tuple where the first element is the list of the function
+    /// parameters and the second one is the return type
+    pub fn deobfuscate_signature(&'s self, signature: &str) -> Option<(Vec<String>, String)> {
```

Comment:
I would rather wrap this into an opaque struct, and then have accessors for the return type and an iterator of arguments.